compute_tools/src/compute.rs

```diff
@@ -644,12 +658,17 @@ impl ComputeNode {
             } else {
                 vec![]
             })
+            .stderr(Stdio::piped())
             .spawn()
             .expect("cannot start postgres process");
 
+        // Start a thread to collect logs from stderr.
```

Comment:
Just adding a `PG:` prefix can be solved by using `log_line_prefix` GUC, so it will be half of what you wanted to achieve in the PR. Although, here you also combine several lines together, iiuc

I'm a bit concerned that now we pump all logs though `compute_ctl`, which should i) increase resource usage; and ii) can it potentially block Postgres if for some reason `compute_ctl` is slow to read them? Can we test that? (or maybe it's clear that it's not a problem even in theory) Otherwise, no real objections to this approach, as you mentioned it may help with adding more context into logs later

Reply:
> Although, here you also combine several lines together, iiuc
`compute_ctl` looks like a better place for adding this prefix, after joining multiline logs.

>  ii) can it potentially block Postgres if for some reason compute_ctl is slow to read them? Can we test that? 
I think blocking is possible, but don't know how to test it or how to make it better. Any ideas?

Reply:
> I think blocking is possible, but don't know how to test it or how to make it better. Any ideas?

Not sure either, but maybe some obscure log level in Postgres like DEBUG5 (iirc), then run some noisy query in a loop

You can then add some simple throttling into compute_ctl, like a 1ms sleep after each log line and test with `pgbench` (it allows custom scripts in just SQL) with and without compute_ctl AND / OR just put a forever blocking in the logs processing and check whether Postgres will work at all

It could be that I'm just overthinking it, so take it with a grain of salt

Reply:
Ok, so I tested it with `sleep(10ms)` before every readline. It is indeed affecting TPS if postgres spams with logs, i.e. your theory that it can block Postgres if `compute_ctl` is slow is correct.

But I don't think it will be a big issue in real life, and I still want to merge it.